packages/flame/lib/src/components/component.dart

```diff
@@ -319,6 +319,28 @@ class Component {
     }
   }
 
+  /// An iterator that performs a deep search on all [children].
```

Comment:
I feel "deep search" is a bit ambiguous: would it search for a particular child and then return it? 
```suggestion
/// Enumerates all children, grand-children, etc. of the current component, in postorder.
```

Reply:
I agree that "deep search" can be seen as ambiguous. Your suggestion is nice but I don't like "etc". What about:

```suggestion
  /// Performs a postorder traversal on all [children].
```

There's no need to specify "of the current component" since referring to the `[children]` variable?
In addition, we already know from the signature that it "Enumerates" since it's an `Iterable`?

Reply:
"Child components" sounds like they are immediate children only. Generally, the first line should give a short summary of what the method does, and then there could be a more elaborate description in the subsequent paragraphs.

Reply:
Indeed, I agree with that since it is detailed in the Effective Dart. I'm running out of ideas. Maybe we can do something like:

1.
```suggestion
  /// Enumerates all nested [children].
  ///
  /// The search is depth-first-search, hence [descendants] are in postorder.
```

2.
```suggestion
  /// Enumerates all [Component]s in the subtree.
  ///
  /// The search is depth-first-search, hence [descendants] are in postorder.
```

Reply:
Maybe?
```dart
/// Enumerates all nested [children].
///
/// The idea is that we move from parent to child until we reach a child without descendants.
/// If we reach such a child we go back (to the point of branching or divergence of paths)
/// and choose another path. This is the case until all the descendants have been traversed.
```

Reply:
I like (1), though there is probably no point in linking `[descendants]`.

Given that the target audience may not be familiar with CS terminology, it would probably be helpful to also add an explanation like @Dan-Crane suggests (but the CS definition must also be present, so that the functionality can be easily understood by experienced programmers as well).

Reply:
@st-pasha can you tell me what `CS` means?🙈

Reply:
@Dan-Crane good question! CS stands for Computer Science.

Reply:
how about `Recursively enumerates...` (that's the word that in my mind immediately clarifies that it is multi-level)

Reply:
I love how `Recursively enumerates all [children]` sounds!

Reply:
done